gensim/similarities/docsim.py

```diff
@@ -456,6 +464,19 @@ def destroy(self):
         for fname in glob.glob(self.output_prefix + '*'):
             logger.info("deleting %s", fname)
             os.remove(fname)
+
+
+def _query_chunk(chunk, index):
+    """ Allow pickling """
+    return list(_query_chunk_gen(chunk, index))
```

Comment:
workers can't return a generators, since generator can't be pickled. This method's sole purpose in life is to allow returning results from a worker thread/process.